ecs-agent/api/resource/ebs_discovery_linux.go

```diff
@@ -0,0 +1,99 @@
+//go:build linux
+// +build linux
+
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package resource
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os/exec"
+	"strings"
+)
+
+// LsblkOutput is used to manage and track the output of `lsblk`
+type LsblkOutput struct {
+	BlockDevies []BD `json:"blockdevices"`
+}
+type BD struct {
+	Name     string    `json:"name"`
+	Serial   string    `json:"serial"`
+	Children []BDChild `json:"children"`
```

Comment:
Is it the children device never going to have their children? I did a quick search, if a logical device is created on the device there can be more layers. Is it not applicable to ebs? Should we let `Children` to be of type `[]*BD` instead?

Reply:
Ah, great point! Will update this.